### DISCUSSION TITLE
Questions about the library

### DISCUSSION THREAD

**Discussion by Redhawk18** [rust-bitcoin/rust-bitcoin#3061] [Q&A]
Hi everyone! I'm trying to start a project of my own to build a crypto miner inspired by xmrig. While monero and randomx are my first goals. Theres a lot to get there with the jsonrpc to communicate with the pool and most have http apis aswell and such. I wanted to ask what is supported here by this library? 

My focus is on more mineable coins like anything based on kawpow, etcash or monero. While I see you only support bitcoin I couldn't find the other required features like how to connect to a pool. Also if you have suggestions for other projects that would help that would be great!

**Comment by Kixunil** [comment]
First of all note that mining Bitcoin is a very difficult business so if you're not doing it for education or fun don't expect much from it.

For connecting to pools the stratum protocol is used and this library does not implement it. You can find an implementation here: https://github.com/braiins/braiins-open/tree/master/protocols/stratum

**Reply by Redhawk18** [reply]
Thanks for the reply I started to check out the crate and I notice it's not as grand as it first looked. Do you have another suggestion or resource for learning?

I found this for the protocol, I'm not sure if it's correct.
https://github.com/aeternity/protocol/blob/master/STRATUM.md

Also my goal currently is to turn how to implement monero not bitcoin because of the higher return, my thought was monero is very widely used and should have a run of the mill algo. But let me know if I should start somewhere else.

**Reply by Kixunil** [reply]
No clue then.
